tests/ui/outer_doc_comment_bang.rs

```diff
@@ -0,0 +1,43 @@
+// run-rustfix
+#![allow(unused)]
+#![warn(clippy::outer_doc_comment_bang)]
+
+//! Real module documentation.
+///! Fake module documentation.
+fn baz() {}
+
+pub mod foo {
+    ///! This module contains useful functions.
+
+    pub fn bar() {}
+}
+
+pub mod foo2 {
+    //! This module contains useful functions.
+
+    pub fn bar() {}
+}
+
+pub mod foo3 {
+    /**! This module contains useful functions.
+     */
+
+    pub fn bar() {}
+}
+
+pub mod foo4 {
+    /*! This module contains useful functions.
+     */
+
+    pub fn bar() {}
+}
+
+pub mod foo5 {
+    ///! This module
+    ///! contains
+    ///! useful functions.
+
+    pub fn bar() {}
+}
+
+fn main() {}
```

Comment:
Can we add a case where the outer doc comment bang is not in the first line, and it's not contiguous?

i.e.

```
///!
///
///!
```
and

```
///
///!
///
```

Reply:
Interesting. This is actually a compile error after applying suggestions [[1]](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ea61b61c49e1a3fa86cd865472dc7017) [[2]](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b302c61a683fd3030c8fea8ee7c0ff9a), so presumably the lint should only suggest changing `///!` to `//!` if it isn't preceded by an inline doc comment (i.e. `///` or `/*!`)?

Reply:
Huh, TIL. 

If it's a compile error then I think we don't need to cover it. Maybe it's good to put a comment somewhere that indicates we know about it and deliberately chose not to cover it.

Reply:
Added the comment under "Known problems"